### PR TITLE
[Feature Store] Adding `FixedWindowType` argument  to `get_online_feature_service`

### PR DESCRIPTION
**Pull request by eyalsol**
Signed-off-by: Eyal Salomon <eyals@iguazio.com>

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -156,7 +163,7 @@ def get_online_feature_service(
     :param run_config:   function and/or run configuration for remote jobs/services
```
Comment (dinal) [inline mlrun/feature_store/api.py:163]: add documentation

**Review by dinal — APPROVED** [review]

File: mlrun/feature_store/retrieval/online.py [hunk 2]
```diff
@@ -12,14 +12,29 @@
...
+    try:
+        from storey import FixedWindowType as QueryByKeyFixedWindowType
+    except ImportError as exc:
+        raise ImportError(f"storey not installed, use pip install storey, {exc}")
+
+    if fixed_window_type == FixedWindowType.LastClosedWindow:
+        return QueryByKeyFixedWindowType.LastClosedWindow
+    if fixed_window_type == FixedWindowType.CurrentOpenWindow:
+        return QueryByKeyFixedWindowType.CurrentOpenWindow
+    return None
```
Comment (Hedingber) [inline mlrun/feature_store/retrieval/online.py:33]: Make this a method of the `FixedWindowType` enum itself, so you would do:
`FixedWindowType. CurrentOpenWindow.to_qbk_fixed_window_type()`

**Review by Hedingber — CHANGES_REQUESTED** [review]
one minor comment

File: mlrun/feature_store/feature_vector.py [hunk 3]
```diff
@@ -407,3 +408,19 @@ def to_csv(self, target_path, **kw):
...
+    def to_qbk_fixed_window_type(self):
+        try:
+            from storey import FixedWindowType as QueryByKeyFixedWindowType
+        except ImportError as exc:
+            raise ImportError(f"storey not installed, use pip install storey, {exc}")
+        if self == FixedWindowType.LastClosedWindow:
+            return QueryByKeyFixedWindowType.LastClosedWindow
+        elif self == FixedWindowType.CurrentOpenWindow:
+            return QueryByKeyFixedWindowType.CurrentOpenWindow
+        return None
```
Comment (Hedingber) [inline mlrun/feature_store/feature_vector.py:426]: If some will add a new FixedWindowType and forget to handle it in this function I want it to explode not to fail quietly (causing hard to debug bugs)
```suggestion
        else:
            raise NotImplementedError(
                f"Provided fixed window type is not supported. fixed_window_type={self}"
            )
```

**Review by Hedingber — APPROVED** [review]

**Pull request merged**